### PR TITLE
Include numeric overrides in neo datetime symbols

### PR DESCRIPTION
**Pull request by Manishearth**
Progress on https://github.com/unicode-org/icu4x/issues/3865  (see design in https://github.com/unicode-org/icu4x/issues/3865#issuecomment-1776126704)

### CODE REVIEW

File: components/datetime/src/fields/length.rs [hunk 1]
```diff
@@ -54,6 +56,8 @@ pub enum FieldLength {
     Six,
     /// A fixed size format for numeric-only fields that is at most 127 digits.
     Fixed(u8),
```
Comment (sffc) [inline components/datetime/src/fields/length.rs:58]: Question: are we using `Fixed` anywhere? Why do we have it?

Comment (Manishearth) [reply components/datetime/src/fields/length.rs:58]: yeah i have no idea what this is

File: provider/datagen/src/transform/cldr/datetime/neo.rs [hunk 2]
```diff
@@ -557,17 +557,59 @@ fn months_convert(
...
+        "y=jpanyear" => (Jpnyear, Some(FieldSymbol::Year(fields::Year::Calendar))),
+        _ => panic!("Found unexpected numeric override {numbering_systems}"),
+    };
+
+    pattern.items.for_each_mut(|item| {
+        if let pattern::PatternItem::Field(ref mut field) = *item {
+            // only replace numeric items
+            if field.length != FieldLength::One {
+                return;
+            }
```
[the remark this answers is not in the document]
Comment (sffc) [reply provider/datagen/src/transform/cldr/datetime/neo.rs]: The spec says

> The numbers attribute can be used to explicitly specify a number system to be used for all of the numeric fields in the date format (as in numbers="hebr"), or for a specific field in the date format (as in numbers="d=hanidays"). This attribute overrides any default numbering system specified for the locale.

Need to dig a little deeper for whether "numeric" refers to strictly `M` or also includes other numeric field lengths like `MM`

File: components/datetime/src/fields/length.rs (hunk 1 above)
Comment (Manishearth) [reply components/datetime/src/fields/length.rs:57]: It seems to be something people can supply via skeletons

File: provider/datagen/src/transform/cldr/datetime/neo.rs [hunk 3]
```diff
@@ -582,6 +582,11 @@ fn apply_numeric_overrides(lp: &ca::LengthPattern, pattern: &mut pattern::runtim
         if let pattern::PatternItem::Field(ref mut field) = *item {
             // only replace numeric items
             if field.length != FieldLength::One {
+                assert!(
+                    field.length != FieldLength::TwoDigit || symbol_to_replace.is_some(),
```
Comment (sffc) [inline provider/datagen/src/transform/cldr/datetime/neo.rs:586]: I'd prefer the condition to be stricter

```suggestion
                    field.length != FieldLength::TwoDigit || symbol_to_replace != Some(symbol),
```

**Review by sffc — APPROVED** [review]

**Pull request merged**